proposals/185/partitioning.md

```diff
@@ -0,0 +1,47 @@
+# Partitioning proposal
+
+Partitioning is a concept of dividing datasets into parts based on values of particular columns like date, city, etc. For Apache Hive and Spark partitioning affected physical location of data, as each partition is stored in a separate directory. This serves in a similar way to database indexing as accessing certain partitions requires reading only some of data directories. 
+
+It is a common usecase to have a large dataset containing data collected over years, while most of the processing require only recent data to proceed, like the last day. Based on that, partitioning is an important concept within data processing and deserves proper OpenLineage represantion.
+
+## Dataset partitioning columns
+
+Specifying partition columns is part of a dataset schema. That's why it should extend existing `SchemaDatasetFacet` and enrich it with an array of `partitionFields` containing names of some of the schema fields. Providing a `partitionFields` would give an extra information on how is the dataset organised. `partitionFields` should be collected for both input and output datasets. 
+
+## Partitions being modified and accessed
+
+Although the dataset may contain a plenty of partitions, only a small subset of them is read or written most of the time. However, it is difficult to find the proper balance for collecting such information. For example, providing an information on partitions read, may result in thousands of useless entries sent within a single OL event. On the other hand, if a certain partition is affected by an upstream issue, it is valuable to know it. 
```

Comment:
I've seen that: a job writing a single daily partition required reading 30 daily partitions of another table.
Perhaps this should be limited during an implementation: if there are 100 partitions read, do not include this in OL event.

Reply:
TBH, I don't consider that even remotely as harmful or irritating as the bigger sources of excess information we get now.  And I think we can structure the partition information so that the incremental cost of each partition is small. E.g., following your `PartitionsDatasetFacet` example, something like the following adds ~50 characters for each partition.
```
{ 
    "partitionsRead": 
        [ 
            {"date": "2022-08-01T00:00:00Z", "countryCode": "CA"}
        ]
}
```
100 partitions means ~5KB of data. The big OL messages we get now are sometimes many MBs, so I'm not too worried about an extra KB or two. I think setting a limit is reasonable, but I'd set that limit pretty high.